server/handlers/validation.go

```diff
@@ -80,9 +80,21 @@ func validateUpdate(cs signed.CryptoService, gun string, updates []storage.MetaU
 			return nil, ErrBadTargets{msg: err.Error()}
 		}
 		repo.SetTargets(targetsRole, t)
+	} else {
+		targetsJSON, err := store.GetCurrent(gun, data.CanonicalTargetsRole)
+		if err != nil {
+			return nil, fmt.Errorf("could not load repo for snapshot generation: %v", err)
```

Comment:
I believe the client always pushes a targets.json. We could try and be more intelligent on the client and only push it when necessary, but the (not correct but protective) assumption is that any change will require targets.json to be resigned.

I should assign a better error type, and I just realized they explicitly say "generation" which may not be correct (and could be troubling to some users) so I'll fix that too.